Allow users to confirm and change their email

CarsonHerness opened the pull request:
Changes in this PR:

- Allow users to update their emails with a modal
- Asks users to confirm their email if they haven't already
- Allows users to de-link their account from Google

akampen commented on public/locales/en/administration.json:
```diff
@@ -84,10 +86,12 @@
         "heading": "Manage Your Account",
-        "manageSignInMethodsHeader": "Manage Sign In Methods"
+        "manageSignInMethodsHeader": "Manage Sign In Methods",
+        "sendEmailVerificationEmail": "Send Email Verification Email",
```
Is supposed to just be "Send Verification Email" (it seems like that's what it is in Spanish). Also change the key to "sendVerificationEmail"

akampen commented on src/components/Admin/ManageAccount/ManageAccount.tsx:
```diff
@@ -22,12 +32,40 @@ const ManageAccount: () => JSX.Element = () => {
+          setSendingEmail(false);
+          setVerificationEmailSent(true);
+        })
+        .catch(() => setError('Unable to send confirmation email'));
```
translate error

akampen reviewed (commented):
I changed my email, then didn't verify it (and left the page open). When I got the email that my email had been changed, I selected "I didn't do this, set it back to my previous email" and it told me that it set it back to amberkampen99@gmail.com, but then when I logged back in it said my email was "madelineruos@gmail.com" (the one I changed it to). I still logged in via google with "amberkampen99@gmail.com", so I believe that the google auth had the desired behavior but maybe we need to make sure that when that happens we also update firebase?

CarsonHerness commented:
> I changed my email, then didn't verify it (and left the page open). When I got the email that my email had been changed, I selected "I didn't do this, set it back to my previous email" and it told me that it set it back to [amberkampen99@gmail.com](mailto:amberkampen99@gmail.com), but then when I logged back in it said my email was "[madelineruos@gmail.com](mailto:madelineruos@gmail.com)" (the one I changed it to). I still logged in via google with "[amberkampen99@gmail.com](mailto:amberkampen99@gmail.com)", so I believe that the google auth had the desired behavior but maybe we need to make sure that when that happens we also update firebase?

Okay so I determined that the issue was that if the account was linked to Google, then both emails remained linked to the account and then some weird behavior happened if you signed in with the old and new email (A new account got created, I didn't see which account was replaced). I added a requirement that your account be unlinked from Google in order to update your email, and I added a button to unlink your account from Google.

akampen reviewed (commented):
A few comments:
- now the admin Home Screen as well as the manage account page loads twice, which is distracting
- Once I unlinked my account with google I could not link it back again
- When I first navigate to the manage sensors page the password button flashes from "add password" to "update password"

CarsonHerness commented:
> I changed my email, then didn't verify it (and left the page open). When I got the email that my email had been changed, I selected "I didn't do this, set it back to my previous email" and it told me that it set it back to [amberkampen99@gmail.com](mailto:amberkampen99@gmail.com), but then when I logged back in it said my email was "[madelineruos@gmail.com](mailto:madelineruos@gmail.com)" (the one I changed it to). I still logged in via google with "[amberkampen99@gmail.com](mailto:amberkampen99@gmail.com)", so I believe that the google auth had the desired behavior but maybe we need to make sure that when that happens we also update firebase?

Alright I misread this the first time, but now I believe it's actually fixed. Now if the Firestore user document's email does not match the Firebase Authentication email, it makes a write to the database.

akampen reviewed (approved):
Not a super big deal because there are no negative consequences, but in the Reauthentication modal the show/hide password button is also a submit button, so it also acts as a trigger to reauthenticate.

CarsonHerness commented:
> Not a super big deal because there are no negative consequences, but in the Reauthentication modal the show/hide password button is also a submit button, so it also acts as a trigger to reauthenticate.

Talked offline, this seems to be unrelated to this PR and possibly an Apple-related bug since I cannot duplicate the issue on my machine or phone
